### PR TITLE
Add metadata/attributes to the created pyramid

### LINKED ISSUE #30: Ensure pyramids have the right attributes/metadata
Over in https://github.com/carbonplan/cmip6-downscaling/blob/main/cmip6_downscaling/methods/common/tasks.py, we have a utility function that is used to post-process the produced pyramid. This function is used to add metadata/attributes to the pyramid. We've talked about including this function in `ndpyramid` directly. I'm opening this issue for clarity/future reference. 

```python
def _pyramid_postprocess(dt: dt.DataTree, levels: int, other_chunks: dict = None) -> dt.DataTree:
    '''Postprocess data pyramid
    Adds multiscales metadata and sets Zarr encoding
    Parameters
    ----------
    dt : dt.DataTree
        Input data pyramid
    levels : int
        Number of levels in pyramid
    other_chunks : dict
        Chunks for non-spatial dims
    Returns
    -------
    dt.DataTree
        Updated data pyramid with metadata / encoding set
    '''
    chunks = {"x": PIXELS_PER_TILE, "y": PIXELS_PER_TILE}
    if other_chunks is not None:
        chunks.update(other_chunks)

    for level in range(levels):
        slevel = str(level)
        dt.ds.attrs['multiscales'][0]['datasets'][level]['pixels_per_tile'] = PIXELS_PER_TILE

        # set dataset chunks
        dt[slevel].ds = dt[slevel].ds.chunk(chunks)
        if 'date_str' in dt[slevel].ds:
            dt[slevel].ds['date_str'] = dt[slevel].ds['date_str'].chunk(-1)

        # set dataset encoding
        dt[slevel].ds = set_zarr_encoding(
            dt[slevel].ds, codec_config={"id": "zlib", "level": 1}, float_dtype="float32"
        )
        for var in ['time', 'time_bnds']:
            if var in dt[slevel].ds:
                dt[slevel].ds[var].encoding['dtype'] = 'int32'

    # set global metadata
    dt.ds.attrs.update({'title': 'multiscale data pyramid'}, **get_cf_global_attrs(version=version))
    return dt
```

Cc @jhamman, @norlandrhagen

### PR DESCRIPTION
**Pull request by andersy005**
- [x] Add `_add_metadata_and_zarr_encoding()`
- [x] Add tests
- [x] Update docstrings
- [x] Fixes #30

### CODE REVIEW

File: ndpyramid/utils.py [hunk 1]
```diff
@@ -29,3 +51,107 @@ def multiscales_template(
...
+        da.encoding['_FillValue'] = default_fillvals.get(
+            da.dtype.str[-2:], None
+        )  # TODO: handle date/time types
+
+        ds[k] = da
+
+    return ds
+
+
+def _add_metadata_and_zarr_encoding(
```
Comment (jhamman) [inline ndpyramid/utils.py:112]: I think we can make this a public utility function

```suggestion
def add_metadata_and_zarr_encoding(
```

File: ndpyramid/utils.py [hunk 2]
```diff
@@ -29,3 +51,107 @@ def multiscales_template(
...
+        chunks.update(other_chunks)
+
+    for level in range(levels):
+        slevel = str(level)
+        pyramid.ds.attrs['multiscales'][0]['datasets'][level]['pixels_per_tile'] = pixels_per_tile
+
+        # set dataset chunks
+        pyramid[slevel].ds = pyramid[slevel].ds.chunk(chunks)
+        if 'date_str' in pyramid[slevel].ds:
+            pyramid[slevel].ds['date_str'] = pyramid[slevel].ds['date_str'].chunk(-1)
```
Comment (jhamman) [inline ndpyramid/utils.py:145]: This should probably live elsewhere. We were using this as a workaround for a specific application. 

```suggestion
```

File: ci/environment.yml [hunk 3]
```diff
@@ -7,6 +7,7 @@ dependencies:
   - esmpy>=8.2.0
   - mpich
   - netcdf4
+  - numba
```
Comment (jhamman) [inline ci/environment.yml:10]: I'm curious why `numba` is required here.

Comment (andersy005) [reply ci/environment.yml:10]: this issue might be relevant to this https://github.com/pangeo-data/xESMF/issues/158

Comment (andersy005) [reply ci/environment.yml]: removed it and everything seems to work fine...

**Review by jhamman — APPROVED** [review]

**Pull request merged**